test/unit/ui.js

```diff
@@ -1,124 +1,215 @@
 import ShopifyBuy from '../../src/buybutton';
 import UI from '../../src/ui';
 import Product from '../../src/components/product';
+import Tracker from '../../src/utils/track';
+import ProductSet from '../../src/components/product-set';
 import Cart from '../../src/components/cart';
+import CartToggle from '../../src/components/toggle';
 import shopFixture from '../fixtures/shop-info';
 
+const DATA_ATTRIBUTE = 'data-shopify-buy-ui';
+
 describe('ui class', () => {
   let ui;
   let script;
 
   let client;
+  let integrations;
   let config;
-  const productConfig = {
-    id: 123,
-    options: {}
-  };
 
   beforeEach(() => {
+    integrations = {
+      errorReporter: {notifyException: sinon.spy()},
+      tracker: 'test',
+    };
     config = {
       domain: 'buckets-o-stuff.myshopify.com',
       storefrontAccessToken: 123,
     };
     client = ShopifyBuy.buildClient(config);
+    client.config.domain = 'test-domain.myshopify.com';
     sinon.stub(client.shop, 'fetchInfo').resolves(shopFixture);
-    ui = new UI(client, {});
     script = document.createElement('script');
-    script.setAttribute('data-shopify-buy-ui', true);
+    script.setAttribute(DATA_ATTRIBUTE, true);
     document.body.appendChild(script);
   });
 
   afterEach(() => {
-    ui = null;
     document.body.removeChild(script);
   });
 
   describe('constructor', () => {
-    it('uses the client.config.domain instead of making a network call', () => {
-      const shopInfo = sinon.spy(client.shop.fetchInfo);
-      client.config.domain = 'test-domain.myshopify.com';
-      ui = new UI(client, {});
+    let resizeSpy;
+    let hostClickSpy;
+    let escSpy;
+    let postMessageSpy;
+    let appendStyleTagSpy;
+
+    beforeEach(() => {
+      resizeSpy = sinon.spy(UI.prototype, '_bindResize');
+      hostClickSpy = sinon.spy(UI.prototype, '_bindHostClick');
+      escSpy = sinon.spy(UI.prototype, '_bindEsc');
+      postMessageSpy = sinon.spy(UI.prototype, '_bindPostMessage');
+      appendStyleTagSpy = sinon.spy(UI.prototype, '_appendStyleTag');
+      ui = new UI(client, integrations, 'test');
+    });
+
+    afterEach(() => {
+      ui = null;
+      resizeSpy.restore();
+      hostClickSpy.restore();
+      escSpy.restore();
+      postMessageSpy.restore();
+      appendStyleTagSpy.restore();
+    });
+
+    it('sets client and domain from passed in client and client.config.domain', () => {
```

Comment:
We're using `params` to refer to the constructor's parameters in a subsequent test's name (`sets stylesOverrides from params`). Could we use that language here as well for consistency?